### DISCUSSION TITLE
Return value after null checking

### DISCUSSION THREAD

**Discussion by HarshNarayanJha** [dotnet/csharplang#5847] [Language Ideas] [answered]
Let's say I do not want to `throw` a `ArgumentNullException`, if the argument is null... Like I don't want to break the code, just return silently from the method, or do a `Debug.Log()`, like in Unity.
Then we have to rely on the old method of null checking.., right?
Won't it negatively affect the code (because I almost completely use C# in Unity)

Anyways, other improvements are great, I really liked the List Patterns feature.

Thanks and keep them up!

**Comment by jnm2** [comment, marked as answer]
That's correct. The C# 11 `!!` feature is specifically to mitigate the hefty amount of existing boilerplate which follows the ArgumentNullException pattern.

Can you clarify your question about what would negatively affect the code?

**Reply by HarshNarayanJha** [reply]
Oh! don't worry.... I was not saying that in a negative sense (I should had used some other word...)
What I meant was that using `!!` inly provides `ArgumentNullException` boilerplate. But if we wanted to use something other than that like `Debug.Log()`, then we will have to rely on old method....
Anyways, great improvements, and keep them up.
